### PR TITLE
GM: Move more interface attributes into CarSpecs

### PR DESCRIPTION
**Pull request by nworb-cire**
**Description**

Moved the following fields into the CarSpecs:
- `minEnableSpeed` for cars that do not fit into the existing 7mph/10kph categories
- `tireStiffnessFactor`
- `steerActuatorDelay`, newly added to base `CarSpecs` class. I put 0.1 as a hopefully sensible default.

Additionally, I set the default lateral tune to torque since only two cars still use PID.

This has the overall effect of heavily reducing most of the GM interface, except for the Volt and Escalade ESV which need to specify the PID tune, and the Silverado min enable speed logic.

**Verification**

CI testing

### CODE REVIEW

File: selfdrive/car/__init__.py [hunk 1]
```diff
@@ -257,6 +257,7 @@ class CarSpecs:
   minSteerSpeed: float = 0.0  # m/s
   minEnableSpeed: float = -1.0  # m/s
   tireStiffnessFactor: float = 1.0
+  steerActuatorDelay: float = 0.0  # seconds
```
Comment (sshane) [inline selfdrive/car/__init__.py:260]: Make sure to set this in interfaces

File: selfdrive/car/gm/values.py [hunk 2]
```diff
@@ -84,82 +85,91 @@ class GMPlatformConfig(PlatformConfig):
...
     GMCarInfo("Chevrolet Volt 2017-18", min_enable_speed=0, video_link="https://youtu.be/QeMCN_4TFfQ"),
-    CarSpecs(mass=1607, wheelbase=2.69, steerRatio=17.7, centerToFrontRatio=0.45),
+    GMCarSpecs(mass=1607, wheelbase=2.69, steerRatio=17.7, centerToFrontRatio=0.45, tireStiffnessFactor=0.469,
+               steerActuatorDelay=0.2, minEnableSpeed=18 * CV.MPH_TO_MS, minSteerSpeed=7 * CV.MPH_TO_MS),
   )
   CADILLAC_ATS = GMPlatformConfig(
     "CADILLAC ATS Premium Performance 2018",
     GMCarInfo("Cadillac ATS Premium Performance 2018"),
-    CarSpecs(mass=1601, wheelbase=2.78, steerRatio=15.3),
+    GMCarSpecs(mass=1601, wheelbase=2.78, steerRatio=15.3, minEnableSpeed=18 * CV.MPH_TO_MS, minSteerSpeed=7 * CV.MPH_TO_MS),
```
Comment (sshane) [inline selfdrive/car/gm/values.py:111]: Can you keep the relevant comments?

**Comment by sshane** [comment]
Can you rebase?

File: selfdrive/car/gm/values.py [hunk 3]
```diff
@@ -89,36 +90,48 @@ class GMPlatformConfig(PlatformConfig):
   dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict('gm_global_a_powertrain_generated', 'gm_global_a_object', chassis_dbc='gm_global_a_chassis'))
 
 
+@dataclass(frozen=True, kw_only=True)
+class GMCarSpecs(CarSpecs):
+  tireStiffnessFactor: float = 0.444
+  steerActuatorDelay: float = 0.1
+  minEnableSpeed: float = 5 * CV.KPH_TO_MS
+  minSteerSpeed: float = 10 * CV.KPH_TO_MS
```
Comment (sshane) [inline selfdrive/car/gm/values.py:98]: already defined

Comment (nworb-cire) [reply selfdrive/car/gm/values.py:98]: These are listed here to set default values for the GM platform

Comment (nworb-cire) [reply selfdrive/car/gm/values.py:98]: Oh sorry, I thought you were referring to the fields, not the class definition. Looks like an incorrect merge; will fix.

**Comment by sshane** [comment]
There's a lot to verify here, can you split it out? Like switching the torque vs. pid tune setting and the min steer/engage speeds?

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**